### PR TITLE
Add MessageEncrypter and Search Functionality

### PR DESCRIPTION
**Pull request by GMOkeefe**
Surrounding functionality still necessary.
- Dynamic update of search page
- Method to pass data from login page to messaging page

Issues #35 & #54

### CODE REVIEW

File: sound_byte/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -20,7 +20,6 @@
 		97C146FC1CF9000F007C117D /* Main.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FA1CF9000F007C117D /* Main.storyboard */; };
```
Comment (hondarider29) [inline sound_byte/ios/Runner.xcodeproj/project.pbxproj]: this file should be deleted and added to gitignore

File: sound_byte/lib/helperClasses/messageEncrypter.dart [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import 'package:encrypt/encrypt.dart';
```
Comment (hondarider29) [inline sound_byte/lib/helperClasses/messageEncrypter.dart]: Could use some more documentation

File: sound_byte/lib/pages/friendScreen.dart [hunk 3]
```diff
@@ -74,7 +77,12 @@ class _FriendScreenState extends State<FriendScreen> {
               Container(
                 child: TextField(
                   onChanged: (String input) {
-                    //TODO: update the list based on input
+                    Firestore.instance.collection('users').where("name", isEqualTo: input).getDocuments().then((querySnapshot)
```
Comment (hondarider29) [inline sound_byte/lib/pages/friendScreen.dart]: The firestore access is dangerous as id the searchedUser is used before the then executes it will error

**Comment by GMOkeefe** [comment]
now adds friend button functionality

File: sound_byte/lib/pages/friendScreen.dart (hunk 3 above)
Comment (hondarider29) [inline sound_byte/lib/pages/friendScreen.dart]: 'users' should be 'Users'

**Pull request merged**